v3/src/components/graph/scatterdots.tsx

```diff
@@ -11,159 +12,135 @@ export const ScatterDots = memo(function ScatterDots(props: {
+    [forceRefreshCounter, setForceRefreshCounter] = useState(0)
 
   const onDragStart = useCallback((event: MouseEvent) => {
-
-      const selectPoint = (iData: worldData[]) => {
-        iData.forEach((datum) => {
-          if (datum.id === tItsID && !datum.selected) {
-            datum.selected = true
-            setHighlightCounter(prevCounter => ++prevCounter)
-          }
-        })
-        return iData
-      }
-
       if (firstTime) {
         setFirstTime(false) // We don't want to animate points until end of drag
       }
       target.current = select(event.target as SVGSVGElement)
-      const tItsID = Number(target.current.property('id'))
+      const tItsID: string = target.current.property('id')
       if (target.current.node()?.nodeName === 'circle') {
         target.current.transition()
           .attr('r', dragRadius)
         setDragID(() => tItsID)
         currPos.current = {x: event.clientX, y: event.clientY}
-        setData(selectPoint(data))
-      }
-      data.forEach(datum => {
-        if (datum.selected) {
-          selectedDataObjects.current[datum.id] = {x: datum.x, y: datum.y}
-        }
-      })
 
-      target.current.classed('dragging', true)
-    }, [setData, data, setHighlightCounter, firstTime, setFirstTime]),
+        worldDataRef.current?.selectCases([tItsID])
+        // Record the current values so we can change them during the drag and restore them when done
+        worldDataRef.current?.selection.forEach(anID => {
+          selectedDataObjects.current[anID] = {
+            x: worldDataRef.current?.getNumeric(anID, xAttrID) ?? 0,
+            y: worldDataRef.current?.getNumeric(anID, yAttrID) ?? 0
+          }
+        })
+      }
+    }, [firstTime, setFirstTime, xAttrID, yAttrID, worldDataRef]),
 
     onDrag = useCallback((event: MouseEvent) => {
-      if (dragID >= 0) {
+      if (dragID !== '') {
         const newPos = {x: event.clientX, y: event.clientY},
           dx = newPos.x - currPos.current.x,
           dy = newPos.y - currPos.current.y
         currPos.current = newPos
         if (dx !== 0 || dy !== 0) {
           const deltaX = Number(xScale?.invert(dx)) - Number(xScale?.invert(0)),
             deltaY = Number(yScale?.invert(dy)) - Number(yScale?.invert(0))
-          setData((prevData) => {
-              prevData.forEach(datum => {
-                if (datum.selected) {
-                  datum.x += deltaX
-                  datum.y += deltaY
-                }
-              })
-              return prevData
-            }
-          )
+          worldDataRef.current?.selection.forEach(anID => {
+            const currX = worldDataRef.current?.getNumeric(anID, xAttrID),
+              currY = worldDataRef.current?.getNumeric(anID, yAttrID)
+            worldDataRef.current?.setValue(anID, xAttrID, Number(currX ?? 0) + deltaX)
+            worldDataRef.current?.setValue(anID, yAttrID, Number(currY ?? 0) + deltaY)
```

Comment:
Rather than running the risk of changing `NaN`s to numbers in the data, I'd've thought something like this would make more sense:
```
            if (isFinite(currX) && isFinite(currY)) {
              worldDataRef.current?.setValue(anID, xAttrID, Number(currX) + deltaX)
              worldDataRef.current?.setValue(anID, yAttrID, Number(currY) + deltaY)
            }
```
or even:
```
            if (isFinite(currX) && isFinite(currY)) {
              worldDataRef.current?.setCaseValues([{
                __id__: anID, [xAttrID]: currX + deltaX, [yAttrID]: currY + deltaY
              }])
            }
```
which has the advantage of being a single action at the model layer.